### PR TITLE
Bump to 1.19.5

### PR DESCRIPTION
**Pull request by mjmac**
Signed-off-by: Michael MacDonald <mjmac.macdonald@intel.com>

### CODE REVIEW

**Comment by brianjmurrell** [comment]
Generally speaking, doesn't this work need to be done for Ubuntu also?  We do currently at least build on Ubuntu, but I suppose we are not building SDL compliant binaries if we are not also using the most recent GO toolchain for that build?

**Comment by mjmac** [comment]
> Generally speaking, doesn't this work need to be done for Ubuntu also? We do currently at least build on Ubuntu, but I suppose we are not building SDL compliant binaries if we are not also using the most recent GO toolchain for that build?

Yes, I suppose so, if we're going to release `.deb` files to the public. For which release is that going to be a requirement?

**Comment by mjmac** [comment]
@brianjmurrell: Am I doing something wrong? I updated to the latest packaging, but still getting this build failure on Leap15.

**Comment by brianjmurrell** [comment]
> Yes, I suppose so, if we're going to release `.deb` files to the public. For which release is that going to be a requirement?

2.6 is the plan I am working to.

> @brianjmurrell: Am I doing something wrong? I updated to the latest packaging, but still getting this build failure on Leap15.

[ You can jump down to the TL;DR at the bottom if you don't care about the details ]

You may be running into what I consider a bug in Artifactory and it's caching.  It's triggered when different tools (yast vs. dnf for example) that use different metadata files access Artifactory.  We overwhelming use dnf on Leap 15 just so that we are not constantly writing `if $suse; then yast bla bla bla; else dnf bla bla bla; fi` everywhere.  Yes, in a given project it could be abstracted, but even copying that abstraction around to the zillion projects is a pain.  So we just use DNF on SUSE.

Except that we occasionally need to use YAST to bootstrap DNF.

The problem then happens that YAST requests metadata file A.  A is an index pointing at B and C (and D and E and F but YAST only uses A and B).   It then uses the index to fetch files B and C.  So B and C are also fetched and cached along with A.  DNF then comes along, pulls (the cached copy of) A and then tries to pull the D and E that A is pointing at.  But they don't yet exist in the cache because YAST didn't request them.  So when DNF requests them, Artifactory tries to fetch them from upstream.  But by now, (while A was being cached) they (A-F) have been updated and the index "A" in the cache is pointing at D, and Es that are no longer present on the remote because they have been replaced with new copies with new names.  Ultimately the metadata set is incomplete (only A-C exist in the cache) and the remote repo no longer has the copies of D-F that the cached A is referring to.

IMO, Artifactory needs to understand that A is an index, and itself (i.e. before the client (yast, dnf, yum, whatever) gets to requesting any files that A points at), read the index and [pre-]fetch and cache all of the files it points to, when client requests the index (A) so that it's got a full and coherent copy of the metadata.  When Artifactory then gives the client the copy of A, it already has the coherent copies of B-F in the cache and will serve them when the client wants one, rather than waiting for the client to request one of them and only then going to try to find it on the remote server.

I have a ticket open with JFrog about this, but our subscription/license/whatever that we have them does not include any support at all, so they ignore our tickets.

TL;DR; you can just wait a while for the out-of-sync metadata to get aged out of the cache and try just re-run the failed job.

**Comment by mjmac** [comment]
> 2.6 is the plan I am working to.

OK, thanks. I'll plan on adding Ubuntu support once 2.4 work settles down.

> TL;DR; you can just wait a while for the out-of-sync metadata to get aged out of the cache and try just re-run the failed job.

Seems to be OK now, thanks.
